src/components/CompareResults/beta/ResultsTable.tsx

```diff
@@ -64,10 +65,28 @@ function processResults(results: CompareResultsItem[]) {
 
 function ResultsTable(props: ResultsTableProps) {
   const { themeMode } = props;
-  const compareResults: CompareResultsItem[] = useAppSelector(
-    (state: RootState) => state.compareResults.data,
-  );
-  const processedResults = processResults(compareResults);
+  const allRevisionsOption =
+    Strings.components.comparisonRevisionDropdown.allRevisions;
+
+  // const compareResults: CompareResultsItem[] = useAppSelector(
+  //   (state) => state.compareResults.data,
+  // );
+  // const old_processedResults = processResults(compareResults);
+
+  const processedResults = useAppSelector((state) => {
+    const { data } = state.compareResults;
+    const { activeComparison } = state.comparison;
+
+    if (activeComparison === allRevisionsOption) {
+      return processResults(data);
+    } else {
+      const results = data.filter(
+        (rev) => truncateHash(rev.new_rev) === activeComparison,
+      );
+
+      return processResults(results);
+    }
+  });
```

Comment:
This looks like a good selector!
Now the next step is to make it memoized using `createSelector`.

Step 1: extract this selector to a new function exported from the ComparisonSlice. It could be called `selectProcessedResults` for example. Of course `processResults` will need to be moved there too. In the future we will possibly want to move that to separate files but let's keep things simple for now.
Then you can import the function here and use in `useAppSelector`.
(documentation: https://redux.js.org/tutorials/essentials/part-2-app-structure#reading-data-with-useselector)

Step 2: use `createSelector` to actually make it memoized. `createSelector` is implemented by `reselect` but it's reexported by redux-toolkit so you can import it with `import { createSelector } from '@reduxjs/toolkit'`.

Then this can look something like this:
```js
import { createSelector } from '@reduxjs/toolkit';
...
export const selectProcessedResults = createSelector(
  // input simple or memoized selectors (in this case they're simple as they only access a property)
  state => state.compareResults.data,
  state => state.comparison.activeComparison,
  // output computed value
  (data, activeComparison) => {
    if (activeComparison === allRevisionsOption) {
      return processResults(data);
    } else {
      const results = data.filter(
        (rev) => truncateHash(rev.new_rev) === activeComparison,
      );

      return processResults(results);
    }
  }
);
```
The selector will be called for every rendering, but:
* if the state object doesn't change, then the previous memoized output will be returned
* if the state object changes, then the 2 input selectors are called. If their results are equals to the previous call, then the previously memoized output will be returned.
* The computing function will be called only if the state object changes and at least one of the input selector results changes.

This has 2 advantages.
1. If the computation is heavy, then we don't do it when we don't have to do it.
2. This ensures that the object reference stays the same if the inputs stays the same => the component where the selector is used won't rerender for no reason (see the doc for `useSelector` above).

(documentation:
https://react-redux.js.org/api/hooks#using-memoizing-selectors
https://redux.js.org/usage/deriving-data-selectors#writing-memoized-selectors-with-reselect
https://github.com/reduxjs/reselect
)

Reply:
[PCF-288 Add selectProcessedResults with memoization](https://mozilla-hub.atlassian.net/browse/PCF-288)